modules/csg/csg_shape.cpp

```diff
@@ -381,7 +441,7 @@ void CSGShape::_notification(int p_what) {
 			parent->_make_dirty();
 		parent = NULL;
 
-		if (use_collision && is_root_shape()) {
+		if (use_collision && is_root_shape() && root_collision_instance.is_valid()) {
```

Comment:
Fixes a separate issue with a NULL free error that could happen when a CSG shape with use_collision enabled was moved inside another shape:

```
ERROR: free: Invalid ID
   At: modules/bullet/bullet_physics_server.cpp:1536
```